versiondb/tsrocksdb/iterator.go

```diff
@@ -23,8 +23,9 @@ func newRocksDBIterator(source *grocksdb.Iterator, prefix, start, end []byte, is
 		} else {
 			source.Seek(end)
 			if source.Valid() {
-				eoakey := moveSliceToBytes(source.Key()) // end or after key
-				if bytes.Compare(end, eoakey) <= 0 {
+				eoakey := source.Key() // end or after key
+				defer eoakey.Free()
```

Comment:
not sure if defer here since source.Prev will change

Reply:
the memory is either managed by the caller or not, if it's not managed by the caller, then the caller should not free it, vice versa.